core/state/history.go

```diff
@@ -7,27 +7,27 @@ import (
 	"github.com/NethermindEth/juno/db"
 )
 
-var _ StateReader = (*StateHistory)(nil)
+var _ StateReader = (*stateHistory)(nil)
 
 // StateHistory represents a snapshot of the blockchain state at a specific block number.
-type StateHistory struct {
+type stateHistory struct {
```

Comment:
Can we remove this completely? Because:
- This is currently unused
- We're already having another state history
- The implementation is not correct, because it needs to handle `ErrCheckHeadState`